catalog/core/catalog-core-localstorageprovider/src/main/java/org/codice/ddf/catalog/content/impl/StorageProviderImpl.java

```diff
@@ -235,110 +240,128 @@ public DeleteStorageResponse delete(DeleteStorageRequest deleteRequest)
     }
 
     @Override
-    public void commit(String id) throws StorageException {
+    public synchronized void commit(StorageRequest request) throws StorageException {
```

Comment:
The current synchronization approach won't make this class thread-safe. We should either synchronize on the `deletionMap` field (my preference) or synchronize all the methods that access the map.